### PR TITLE
Adapt schema derivation

### PR DESCRIPTION
**Pull request by mbore**

### CODE REVIEW

File: core/src/main/scala-3/sttp/tapir/macros/SchemaMacros.scala [hunk 1]
```diff
@@ -3,8 +3,71 @@ package sttp.tapir.macros
...
+              report.throwError(s"Invalid use of path element $pathEl. $ShapeInfo, got: ${tree}")
+            case Nil =>
+              report.throwError(s"Invalid use of path element(Nil). $ShapeInfo, got: ${tree}")
+          }
+          
+          idents.flatMap(toPath(_, newAcc))
+        }
+
+        /** Wild card from path */
+        case i: Ident =>
```
Comment (adamw) [inline core/src/main/scala-3/sttp/tapir/macros/SchemaMacros.scala:47]: what case is it? maybe you could add example code to the comment

File: core/src/main/scala-3/sttp/tapir/macros/SchemaMacros.scala [hunk 2]
```diff
@@ -3,8 +3,71 @@ package sttp.tapir.macros
...
+    
+    enum PathElement {
+      case TermPathElement(term: String, xargs: String*) extends PathElement
+      case FunctorPathElement(functor: String, method: String, xargs: String*) extends PathElement
+    }
+
+    def toPath(tree: Tree, acc: List[PathElement]): Seq[PathElement] = {
+      def typeSupported(quicklensType: String) =
+        Seq("ModifyEach", "ModifyEither", "ModifyEachMap")
+          .exists(quicklensType.endsWith)
```
Comment (adamw) [inline core/src/main/scala-3/sttp/tapir/macros/SchemaMacros.scala:26]: quicklens? :)

File: core/src/main/scala-3/sttp/tapir/macros/SchemaMacros.scala [hunk 3]
```diff
@@ -3,8 +3,71 @@ package sttp.tapir.macros
...
+          toPath(deep, PathElement.TermPathElement(ident) :: acc)
+        /** Method call with no arguments and using clause */
+        case Apply(Apply(TypeApply(Ident(f), a), idents), b) if typeSupported(f) => {
+           val newAcc = acc match {
+            /** replace the term controlled by quicklens */
+            case PathElement.TermPathElement(term, xargs @ _*) :: rest => PathElement.FunctorPathElement(f, term, xargs: _*) :: rest
+            case pathEl :: _ =>
+              report.throwError(s"Invalid use of path element $pathEl. $ShapeInfo, got: ${tree}")
+            case Nil =>
+              report.throwError(s"Invalid use of path element(Nil). $ShapeInfo, got: ${tree}")
```
Comment (adamw) [inline core/src/main/scala-3/sttp/tapir/macros/SchemaMacros.scala:40]: what case is it? probably error  message could be better

Comment (mbore) [reply core/src/main/scala-3/sttp/tapir/macros/SchemaMacros.scala:42]: hm I just copied it from scala 2 version and haven't found test example for it yet. I will try to add some.

Comment (adamw) [reply core/src/main/scala-3/sttp/tapir/macros/SchemaMacros.scala:42]: ah :) well maybe we could fix both at the same time ;) or maybe this case is impossible

**Pull request merged**